.ci/scripts/diff_dswx_files.py

```diff
@@ -0,0 +1,139 @@
+#   diff_dswx_files.py
+#
+#   For: OPERA project
+#
+#   11-03-2023 - Ray Bambery
+#       initial release
+#
+#   Description: Python program to perform Quality Assurance Test
+#       on a directory of NETcdf4 files - searches for files with
+#       a .tif file extension.
+#       This script calls a subprocess script,  dswx_comparison.py
+#       located in the dswx_s1_x.y.z directory shown as <DSWX-S1_DIR>
+#       in the wiki page 
+#       https://github.com/nasa/opera-sds-pge/wiki/DWSX_S1_Beta-Acceptance-Test-Instruuctions
+#       which ensures there are matching files in each input directory
+##
+#   Arguments:
+#       expected_output_dir output_dir dir
+#
+#   Outputs: A terminal stream of  results of the tests showing 
+#       PASS (in GREEN) or FAIL (in RED)
+#   
+#   Calls:  
+#       dswx_comparison.py
+#
+import os
+import sys
+import subprocess
+import argparse
+import glob
+#import netCDF4 as nc
+
+##################################################################
+def _parse_args():
+    parser = argparse.ArgumentParser(
+        description='2 directories for comparison of files'
+    )
+    parser.add_argument('input dirs',
+        type=str,
+        nargs=2,
+        help='Input dir1 Input_dir2')
+
+    print ("len(sys.argv) = ",len(sys.argv))
+    if len(sys.argv) == 1:
+        parser.print_help()
+#        print ("options are ",options)
+        sys.exit(0)
+    else:
+        if len(sys.argv) == 2:
+            print ("Missing second input directory")
+            sys.exit(-1)
+        else:
+            print ("sys.argv[1] = ",sys.argv[1])
+            print ("sys.argv[2] = ",sys.argv[2])
+#           options = parser.parse_args()
+#           print ("options are ", options)
+
+    return sys.argv
+#################################################################
+def get_files(options):
```

Comment:
This is as good a place as any to describe the standards we use for function docstrings:

As much as possible, I want to see docstrings that follow the "numpy" convention for documentation. The general template for a numpy-style docstring looks like the following:

```python
def function_name(arg1, arg2)
    """
    <The first sentence/paragraph of the docstring should explain the purpose of the function at a high-level.>
 
    Notes
    ------
    <Any additional documentation worth calling out that is not provided by the main descriptive sentence/paragraph>

    Parameters
    ------------
    arg1 : <argument type>
       <Description of the argument, including any constraints on value or invariants upheld by the function>
    arg2: <argument type>
       <Similar description for arg2>

    Raises
    -------
    <Exception type>
         <If the function can raise an exception, document the circumstances that would cause it here>

    Returns
    --------
    result : <type of returned value or values>
         <Description of the returned value or values>
    """
  
```
Note that if the function does not take any parameters, a `Parameters` section is not necessary. Similarly, if the function does not return anything, no `Returns` section is necessary, and if no exceptions are **directly** raised by the function, no `Raises` section is necessary.

For very simple functions, the docstring can be confined to a single line beneath the `def`:

```python
def simple_function():
    """Example single line docstring describing the simple function"""
```

You can find some good examples of our documentation practices within the following modules:
https://github.com/nasa/opera-sds-pge/blob/main/src/opera/util/run_utils.py
https://github.com/nasa/opera-sds-pge/blob/main/src/opera/util/tiff_utils.py

For the sake of practice, I would like to see docstings like the ones I've described added to the `_parse_args` and `get_files` functions within this module. Don't worry about `dswx_comparison.py` since it already has similar(ish) docstrings already.